devito/operator.py

```diff
@@ -82,7 +82,7 @@ def expr_cse(expr):
     for temp, value in temps:
         if isinstance(value, IndexedBase):
             to_revert[temp] = value
-        elif isinstance(value, Add):
+        elif isinstance(value, Add) and not set([t, x, y, z]).isdisjoint(set(value.args)):
```

Comment:
Hard-coding that dimension set is hacky. I'm happy for this to go in now, but it will most certainly create problems in the future.